File: backend-java/app/src/main/java/org/example/Response.java
package org.example;

import java.util.HashMap;

import org.json.simple.JSONObject;

public class Response {
	int statusCode;
	String contentType;
	String body;

	public Response() {
		statusCode = 200;
		contentType = "text/plain";
	}

	public Response(int statusCode, String contentType) {
		this.statusCode = statusCode;
		this.contentType = contentType;
	}

	void setStatusCode(int statusCode) {
		this.statusCode = statusCode;
	}

	void write(String body) {
		this.body = body;
	}

	void setContentType(String contentType) {
		this.contentType = contentType;
	}

	void writeJson(JsonObj map) {
		contentType = "application/json";
		body = map.toJson();
	}
	void writeJson(HashMap<?, ?> json) {
		contentType = "application/json";
		JSONObject object = new JSONObject(json);
		write(object.toJSONString());
	}
}

File: backend-scala/app/src/main/kotlin/org/example/Response.kt
package org.example;

interface MyJsonObject {
	fun toJson(): String
}

class Response(code: Int = 200, contentType: String = "text/plain") {
	var code: Int
	var contentType: String
	var body: String = ""
	init {
		this.code = code
		this.contentType = contentType
	}
	fun write(body: String) {
		this.body = body
	}
	fun writeJson(Object: MyJsonObject) {
		this.body = Object.toJson()
		this.contentType = "application/json"
	}
}